cfgov/unprocessed/apps/filing-instruction-guide/js/fig-sidenav.js

```diff
@@ -39,7 +37,9 @@ const defaultScrollOffset = (target) => {
 const handleMobileNav = (event) => {
   event.preventDefault();
   if (event.target.matches('.o-secondary-nav_link')) {
-    secondaryNav.collapse();
+    // This would ideally be secondaryNav.collapse() but there's a buggy race condition
+    // with that method that makes collapsing unreliable
+    document.querySelector('.o-secondary-nav_header').click();
```

Comment:
Do you have more context for this? I'd like to fix this directly in `.collapse()` if possible.

Reply:
For some reason the `u-max-height-zero` and `u-hidden` classes are unpredictably set on the expandable body. I spent a few hours going down a rabbit hole before realizing everything works perfectly if I just simulate a click.

To reproduce the bug:

1. Visit the [FIG](https://www.consumerfinance.gov/data-research/small-business-lending/filing-instructions-guide/2024-guide/)
2. Resize the screen to a mobile-ish size (and reload the page to be safe)
3. Open the top `In this section` menu
4. Click a random link to be jumped down the page
5. Manually scroll back to the top of the page
6. Try to close and open the `In this section` menu a few times and you'll see that it sometimes doesn't open or close

Reply:
Resolved in https://github.com/cfpb/consumerfinance.gov/pull/7818